src/pages/DirectMessage/hooks/useMessageGroupList.ts

```diff
@@ -0,0 +1,36 @@
+import useAuthUserStore from "@/stores/useAuthUserStore"
+import { Conversation } from "@/types"
+import { useQuery } from "@tanstack/react-query"
+import { useParams } from "react-router-dom"
+import getMessageGroupListAPI from "../apis/getMessageGroupListAPI"
+export const QUERY_KEY_GET_GROUP_MESSAGELIST = "GET_GROUP_MESSAGELIST"
+
+const useMessageGroupList = () => {
+  const { myId } = useAuthUserStore()
+  const { id: userId } = useParams()
+  const { data } = useQuery({
+    queryKey: [QUERY_KEY_GET_GROUP_MESSAGELIST],
+    queryFn: getMessageGroupListAPI,
+    initialData: [],
+    refetchInterval: 1000, // 재 요청
+
+    select: (GroupMessageList: Conversation[]) =>
+      GroupMessageList.map((MessageList) => {
+        // 최근에 보낸 메시지가 내가 보낸거면 채팅을 하고 있거나 읽음 처리
+        if (
+          myId === MessageList.sender._id ||
+          userId === MessageList.sender._id
+        ) {
+          return {
+            ...MessageList,
+            seen: true,
+          }
+        } else {
+          return MessageList
+        }
```

Comment:
[P3] 이 부분 early return으로 바꿀 수 있을 것 같아요! 👍
```suggestion
        if (
          myId === MessageList.sender._id ||
          userId === MessageList.sender._id
        ) {
          return {
            ...MessageList,
            seen: true,
          }
          return MessageList
        
```